dashboard/src/actions/repos.ts

```diff
@@ -162,14 +162,18 @@ export const fetchRepoSecrets = (
     const {
       clusters: { currentCluster },
     } = getState();
-    // TODO(andresmgot): Create an endpoint for returning credentials related to an AppRepository
-    // to avoid listing secrets
-    // https://github.com/kubeapps/kubeapps/issues/1686
-    const secrets = await Secret.list(currentCluster, namespace);
-    const repoSecrets = secrets.items?.filter(s =>
-      s.metadata.ownerReferences?.some(ownerRef => ownerRef.kind === "AppRepository"),
-    );
-    dispatch(receiveReposSecrets(repoSecrets));
+    try {
+      // TODO(andresmgot): Create an endpoint for returning credentials related to an AppRepository
+      // to avoid listing secrets
+      // https://github.com/kubeapps/kubeapps/issues/1686
+      const secrets = await Secret.list(currentCluster, namespace);
+      const repoSecrets = secrets.items?.filter(s =>
+        s.metadata.ownerReferences?.some(ownerRef => ownerRef.kind === "AppRepository"),
+      );
+      dispatch(receiveReposSecrets(repoSecrets));
+    } catch (e) {
+      dispatch(errorRepos(e, "fetch"));
+    }
```

Comment:
This has a caveat. There is a situation in which "it's okay" to fail retrieving repo secrets. If you are in a namespace, this function would be called twice, one for the current namespace and another to retrieve repos in the `kubeapps` namespace (to show global repositories, read-only). In that case, non-privileged users won't be able to retrieve secrets from the `kubeapps` repository but we don't want to fail because that's fine (people need to go to the `kubeapps` namespace to manage global repos).

What I would do, to avoid the unhandled rejection and allow that situation, is adapt `fetchRepos` (line 190) for the case in which several namespaces are given and avoid calling `fetchRepoSecrets` for the kubeapps namespace. Ping me if you need more info about this.

Reply:
Nevermind, I was double-checking this when I found out that `fetchRepos` is luckily not calling `fetchRepoSecrets` with the global namespace (because of the order of the params). This is highly weak but nothing to do with this PR, it can be merged as it is.

Reply:
Great! Thank you for having a look at this issue!! Merging then :)